mantidproject/mantid
# Allow start and end X boxes to be negative in model fitting

**Description of work.**
This PR allows you to enter a negative value for the start or end X when necessary in the model fitting tab. Previously the double was only allowed to be positive (probably because we didn't need to allow negative numbers previously).

**To test:**
Load HIFI 77766-77776. Wait for them to load
Fitting tab, select simultaneous fit
Add a Chebyshev function with n=4 and ExpDecayOsc with Frequency =1
Sequential fit all with Initial parameters for each fit checked
Create results table on the results tab
Go to model fitting. Select the `f1.Phi` parameter to be the x parameter.
Attempt to change the Start X (which is already negative) to another negative number. This should now be possible.
Make sure the Start X doesn't go lower than the lowest data value on the x axis of the plot.

Fixes #32450

*This does not require release notes* because **previously we did not need to enter a negative value for start and end X, but we will for model fitting.*

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
